### PR TITLE
Add support for alert notification channels

### PR DESCRIPTION
**Pull request by runz0rd**

### CODE REVIEW

**Comment by safaci2000** [comment]
First of all thank you for the submission.

1. I think you have some compile errors, if you don't mind addressing those.
2. The folder "integration" tests has some tests that execute against a Docker instance of grafana.  Do you mind adding some basic CRUD tests for the alert notifications you're adding.

I'll do a more in depth review on Monday, but that looks good so far.

**Comment by safaci2000** [comment]
Also, if you need any configuration changes please update the import example file to reflect that.

File: api/filters.go [hunk 1]
```diff
@@ -129,3 +129,20 @@ func (s DatasourceFilter) Validate(items map[string]string) bool {
...
+	return []string{}
+}
+
+//Validate returns true if mapped values are valid
+func (s AlertNotificationFilter) Validate(items map[string]string) bool {
+	if s.GetFilter(Name) == "" {
+		return true
+	}
+	return items[Name] == s.GetFilter(Name)
+}
```
[the remark this answers is not in the document]
Comment (runz0rd) [reply api/filters.go]: removed filter since the original idea was to import/export all alert notification channels

**Comment by safaci2000** [comment]
poke friendly reminder

**Review by safaci2000 — APPROVED** [review]

**Pull request merged**